indexer/feature_data.cpp

```diff
@@ -262,20 +263,31 @@ bool FeatureParams::AddHouseName(string const & s)
   return false;
 }
 
-bool FeatureParams::AddHouseNumber(string const & ss)
+bool FeatureParams::AddHouseNumber(string houseNumber)
 {
-  if (!feature::IsHouseNumber(ss))
+  ASSERT(!houseNumber.empty(), ("This check should be done by the caller."));
+  ASSERT_NOT_EQUAL(houseNumber.front(), ' ', ("Trim should be done by the caller."));
+
+  // Negative house numbers are not supported.
+  if (houseNumber.front() == '-' || houseNumber.find(u8"－") == 0)
```

Comment:
Пользователи могут вводить любую ерунду.

> On 12.4.2016, at 13.34, mpimenov <notifications@github.com> wrote:
> 
> In indexer/feature_data.cpp:
> 
> >  {
> > -  if (!feature::IsHouseNumber(ss))
> > +  ASSERT(!houseNumber.empty(), ("This check should be done by the caller."));
> > +  ASSERT_NOT_EQUAL(houseNumber.front(), ' ', ("Trim should be done by the caller."));
> > +
> > +  // Negative house numbers are not supported.
> > +  if (houseNumber.front() == '-' || houseNumber.find(u8"－") == 0)
> Ок.
> Странно, что такое встречается, и что от этого нужно как-то защищаться.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>